util/json.c

```diff
@@ -38,7 +38,7 @@ struct json_object *util_json_object_new_uint64(uint64_t i)
 struct json_object *util_json_object_new_uint128(nvme_uint128_t  val)
 {
 	struct json_object *obj;
-	obj = json_object_new_string(uint128_t_to_string(val));
+	obj = json_object_new_string(uint128_t_to_string_with_sep(val, NULL));
```

Comment:
I think we need to introduce a new json object for the type uint128 and write our own serializer (see  `json_object_set_serializer()`, https://github.com/json-c/json-c/blob/master/tests/test_set_serializer.c) , because this will still generates a string and not a number.

Reply:
`util_json_object_new_double()` and `util_json_object_new_uint64() ` are fallbacks implementation which cheat a bit and generate just a string. But it supposed to generate actual numbers. 

```c
					json_object_add_value_uint64(jns, "UsedBytes", nuse);
					json_object_add_value_uint64(jns, "MaximumLBA", nvme_ns_get_lba_count(n));
					json_object_add_value_uint64(jns, "PhysicalSize", nsze);
```

gives

```json
      "UsedBytes":512110190592,
      "MaximumLBA":1000215216,
      "PhysicalSize":512110190592,
```

with a current version of json-c. The type is supposed to be a number and not string (in JSON spec terms). The reason why it works for the bug reported in #1827 is that he uses Python which is able to switch between types very easy. It is a bug since introduction of the uint128 type.

Reply:
Thanks for your explanations. I could understand and just pushed the patch fixed.